server/emails/index.js

```diff
@@ -0,0 +1,195 @@
+/**
+ * Mail templates.
+ *
+ * @module server/emails
+ */
+const glob = require('fast-glob');
+const { compile: compileHandlebars } = require('handlebars');
+const { isEmpty, isString, mapValues, pick } = require('lodash');
+const { basename: getBasename, dirname: getDirname, extname: getExtension, join: joinPath, relative: getRelativePath } = require('path');
+
+const config = require('../../config');
+const { loadFileWithFrontmatter } = require('../utils/frontmatter');
+
+const logger = config.logger('emails');
+
+const emails = {};
+const emailsDir = config.path('server', 'emails');
+
+/**
+ * Returns an object containing the subject and body (text and optionally HTML)
+ * for the specified email.
+ *
+ * @param {string} name - The email identifier, corresponding to a subdirectory in `server/emails`.
+ * @param {string} locale - The locale to use to select the templates in the correct language.
+ * @param {Object} [templateData={}] - Data to pass to the compiled Handlebars template function.
+ * @returns {EmailOptions} Nodemailer options for the specified email.
+ */
+exports.buildEmailOptions = function(name, locale, templateData = {}) {
+  if (!emails[name]) {
+    throw new Error(`No email templates named "${name}" (looking in ${joinPath(emailsDir, name)})`);
+  } else if (!emails[name][locale]) {
+    throw new Error(`NO email templates named "${name}" are available for locale "${locale}" (looking in ${joinPath(emailsDir, name)})`);
```

Comment:
Be less agressive...
```suggestion
    throw new Error(`No email templates named "${name}" are available for locale "${locale}" (looking in ${joinPath(emailsDir, name)})`);
```